fv/test-connection/test-connection.go

```diff
@@ -78,7 +82,15 @@ func main() {
 	ipAddress := arguments["<ip-address>"].(string)
 	port := arguments["<port>"].(string)
 	sourcePort := arguments["--source-port"].(string)
-	log.Infof("Test connection from %v:%v to IP %v port %v", namespacePath, sourcePort, ipAddress, port)
+	// Set default for source IP.
+	sourceIpAddress := defaultSourceIP
+	if strings.Contains(ipAddress, ":") {
+		sourceIpAddress = "::"
+	}
+	if srcIP, ok := arguments["--source-ip"].(string); ok {
```

Comment:
I *think* that because you have `[default 0.0.0.0]` in the usage, you'll get `0.0.0.0` for `arguments["--source-ip"]` even if it wasn't on the command line.  That means this if-test always fires, and you klobber the IPv6 default of `::`.

However, I'm also pretty sure that you can use `::` as the source address even if you're connecting to an IPv4 remote and Linux will pick a local IPv4 address on the correct interface automatically.  So the upshot is that you can simplify all this defaulting code by setting `[default: ::]` in the usage and then just

```
sourceIpAddress := arguments["--source-ip"].(string)
```

like most of the other args do.

Reply:
Thanks, I fixed this up a bit: bb800ef41057daaa8c7e9403fc2b67e73d63e4c9

I tried using `::` as the default but I ended up getting an error in `net.ResolveTCPAddr`:

```
level=info msg=\"Resolved tcp addr\" addr= resolved=\":0\"\npanic: address 10.65.0.2:8055: unexpected address type
```

It looks like the local address (`::`) for some reason was not parsable as an addr? https://github.com/golang/go/blob/master/src/net/dial.go#L592
I'm not really sure what's going on but I can go back and take a look later.